mailbox/plugin/quota-mailing/src/main/java/org/apache/james/eventsourcing/CommandDispatcher.java

```diff
@@ -0,0 +1,77 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.eventsourcing;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.IntStream;
+
+import com.google.common.collect.ImmutableList;
+
+public class CommandDispatcher {
+
+    private static final int MAX_RETRY = 10;
+
+    public interface Command {
+    }
+
+    public interface CommandHandler<C extends Command> {
+        Class<C> getCommandClass();
+
+        List<? extends Event> handle(C c);
+    }
+
+    private final EventBus eventBus;
+    private final Map<Class, CommandHandler> handlers;
+
+    public CommandDispatcher(EventBus eventBus) {
+        this.eventBus = eventBus;
+        this.handlers = new ConcurrentHashMap<>();
+    }
+
+    public CommandDispatcher register(CommandHandler commandHandler) {
+        handlers.put(commandHandler.getCommandClass(), commandHandler);
+        return this;
+    }
+
+    public void dispatch(Command c) {
+        IntStream.range(0, MAX_RETRY)
```

Comment:
maybe we could split the code to write : 
```
trySeveralTimes(tryDispatch).orElseThrow(RuntimeException::new)
```
BTW, we could create a named exception for that case also